File: code/python/day01/part2.py
#!/usr/bin/env python

import argparse, os, sys
import numpy as np

from part1 import load_data, calc_deltas, count_upside_measurements

def smooth_by_three(data:np.ndarray) -> np.ndarray:
    assert(isinstance(data, np.ndarray))
    assert(len(data) > 2)

    a = list()

    for i in range(2, len(data)):
        a.append(data[i-2] + data[i-1] + data[i])
    return np.array(a)

def main():

    parser = argparse.ArgumentParser(description='Day 1, Part 2')
    parser.add_argument('input_filename', type=str, help='Input text')
    args = parser.parse_args()

    data = load_data(args.input_filename)
    smooth_data = smooth_by_three(data)
    deltas = calc_deltas(smooth_data)
    upside_swings = count_upside_measurements(deltas)
    
    print(f'Data:          \t{data}')
    print(f'Smoothed Data: \t{smooth_data}')
    print(f'Smooth Deltas: \t{deltas}')
    print(f'Upside Swings: \t{upside_swings}')


if __name__ == '__main__':
    main()

File: code/go/day01/part2.go
package main

import (
	"fmt"
	utils "./utils"
)

func main() {

	input_filename := utils.Get_args()
	data := utils.Load_data(input_filename)
	smooth_data := utils.Smooth_data(data)
	deltas := utils.Calculate_deltas(smooth_data)
	upsides := utils.Count_upside(deltas)

	fmt.Printf("Data: %v\n", data)
	fmt.Printf("Smooth Data: %v\n", smooth_data)
	fmt.Printf("Deltas: %v\n", deltas)
	fmt.Printf("Upsides: %v\n", upsides)
}